components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.cache.ui/src/main/java/org/wso2/carbon/mediator/cache/ui/CacheMediator.java

```diff
@@ -21,439 +21,513 @@
+     *
+     * @param maxMessageSize maximum size of the messages to be set in bytes.
+     */
+    public void setMaxMessageSize(int maxMessageSize) {
+        this.maxMessageSize = maxMessageSize;
+    }
+
+    /**
+     * This method gives the local name of the mediator.
+     *
+     * @return local name of mediator.
+     */
+    public String getTagLocalName() {
+        return CachingConstants.CACHE_LOCAL_NAME;
+    }
+
+    /**
+     * Creates XML representation of the cache mediator as an OMElement
+     *
+     * @param parent OMElement which take child as created OMElement
+     */
+
+    /**
+     * This method gives the HTTP method that needs to be cached
+     *
+     * @return the HTTP method to be cached
+     */
+    public String getHTTPMethodsToCache() {
+        return hTTPMethodsToCache;
+    }
+
+    /**
+     * This sets the HTTP method that needs to be cached
+     *
+     * @param hTTPMethodToCache the HTTP method to be cached
+     */
+    public void setHTTPMethodsToCache(String hTTPMethodToCache) {
+        this.hTTPMethodsToCache = hTTPMethodToCache;
+    }
+
+    /**
+     * @return the protocol type of the messages
+     */
+    public String getProtocolType() {
+        return protocolType;
+    }
+
+    /**
+     * This method sets protocolType of the messages.
+     *
+     * @param protocolType protocol type of the messages.
+     */
+    public void setProtocolType(String protocolType) {
+        this.protocolType = protocolType;
+    }
+
+    /**
+     * @return The regex expression of the HTTP response code of the messages to be cached
+     */
+    public String getResponseCodes() {
+        return responseCodes;
+    }
+
+    /**
+     * This method sets the response codes that needs to be cached.
+     *
+     * @param responseCodes the response codes to be cached in regex form.
+     */
+    public void setResponseCodes(String responseCodes) {
+        this.responseCodes = responseCodes;
+    }
+
+    /**
+     * This method gives array of headers that would be excluded when hashing.
+     *
+     * @return array of headers to exclude from hashing
+     */
+    public String getHeadersToExcludeInHash() {
+        return headersToExcludeInHash;
+    }
+
+    /**
+     * This method sets the array of headers that would be excluded when hashing
+     *
+     * @param headersToExcludeInHash array of headers to exclude from hashing.
+     */
+    public void setHeadersToExcludeInHash(String headersToExcludeInHash) {
+        this.headersToExcludeInHash = headersToExcludeInHash;
+    }
+
+    public OMElement serialize(OMElement parent) {
+        OMElement cache = fac.createOMElement(CachingConstants.CACHE_LOCAL_NAME, synNS);
+        saveTracingState(cache, this);
+
+        if (id != null) {
+            cache.addAttribute(fac.createOMAttribute("id", nullNS, id));
+        }
+
+        if (collector) {
+            cache.addAttribute(fac.createOMAttribute("collector", nullNS, "true"));
```

Comment:
Defining Qname seems waste of instructions, since createOMAttribute function doesn't overloaded function to accept QName. let's create string constants instead